File: 22. Check identical trees.cpp
#include <bits/stdc++.h>
using namespace std;
struct Node
{
    int data;
    Node* left;
    Node* right;
    Node(int d) : data(d), left(nullptr), right(nullptr) {}
};
bool identical(Node* a, Node* b)
{
    if (!a && !b)
        return true;
    if (!a || !b)
        return false;
    return a->data == b->data && identical(a->left, b->left) && identical(a->right, b->right);
}
int main()
{
    Node* a = new Node(1);
    a->left = new Node(2);
    a->right = new Node(3);
    Node* b = new Node(1);
    b->left = new Node(2);
    b->right = new Node(3);
    cout << (identical(a, b) ? "Identical" : "Not Identical") << endl;
    return 0;
}

File: 22. Check identical trees.py
class Node:
    def __init__(self,data): self.data=data; self.left=None; self.right=None

def identical(a,b):
    if not a and not b: return True
    if not a or not b: return False
    return a.data==b.data and identical(a.left,b.left) and identical(a.right,b.right)

if __name__=='__main__':
    a=Node(1); a.left=Node(2); a.right=Node(3)
    b=Node(1); b.left=Node(2); b.right=Node(3)
    print('Identical' if identical(a,b) else 'Not Identical')
